### PR TITLE
WIP: add libgd

### PR DESCRIPTION
**Pull request by croth1**
Initial try - still having problems with `make check` locally. Let's see what ci thinks.

### CODE REVIEW

File: recipes/libgd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  url: https://github.com/libgd/libgd/releases/download/gd-{{version}}/libgd-{{ version }}.tar.gz
+  patches:
+    - osx_version.patch
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - autoconf
```
Comment (jakirkham) [inline recipes/libgd/meta.yaml]: Looks like Mac want `automake`. We have it so feel free to add it here.

Comment (croth1) [inline recipes/libgd/meta.yaml]: yep, but it's already late here - will continue tomorrow 👍

**Comment by croth1** [comment]
Running `make check` is tricky, because files generated during testing have to be patched. My current solution tests in two passes on OS X: In the first pass all tests fail, but all testing files are generated. The second pass then patches and runs the actual test suite. Not particularly nice, but seems to work.

**Comment by ocefpaf** [comment]
> My current solution tests in two passes on OS X: In the first pass all tests fail, but all testing files are generated. The second pass then patches and runs the actual test suite. Not particularly nice, but seems to work.

That seems fine. I don't know if there is a better way...

The recipe LGTM. Our master is failing right now. So let's wait until we fix that to merge this one and avoid backlogging the feedstock creation.

File: recipes/libgd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libgd.a      # [linux]
+    - test -f $PREFIX/lib/libgd.dylib  # [osx]
+    - test -f $PREFIX/lib/libgd.la
+about:
+    home: http://libgd.github.io/
+    summary: library for the dynamic creation of images
+    license: Custom OSS
```
Comment (ocefpaf) [inline recipes/libgd/meta.yaml]: According to [this](https://github.com/libgd/libgd/issues/47) it should be `GD license`. You can also add a `license_file` metadata pointing to the file in the source dir. See [zlib](https://github.com/conda-forge/zlib-feedstock/blob/master/recipe/meta.yaml) for an example.

**Comment by croth1** [comment]
For now I didn't include webp support. We can of course enable it when webp makes it into conda-forge. I think I am happy now. 👍

File: recipes/libgd/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+./configure --prefix=$PREFIX \
+            --without-xpm \
+            --without-x \
+            --disable-werror \
+            --without-webp \
+|| { cat config.log; exit 1; }
+
+make && make install
+make check || failed=1
```
Comment (ocefpaf) [inline recipes/libgd/build.sh]: Please add that comment you made about the patch here.

**Comment by ocefpaf** [comment]
Thanks @croth1 ! I made a minor comment, but you can address that in the feedstock as this is good to go.

**Pull request merged**